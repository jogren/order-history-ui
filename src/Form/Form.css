form {
  display: flex;
  flex-direction: column;
  width: 250px;
}

input {
  height: 25px;
  margin-bottom: 7px;
  font-size: 12px;
}

.Form_button {
  height: 28px;
  margin-bottom: 7px;
  background-color: lightblue;
  font-size: 14px;
}
