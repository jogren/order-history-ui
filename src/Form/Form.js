import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      description: '',
      img: '',
      price: '',
      id: Date.now()
    }
  }

  updateInputState = e => {
    this.setState({ [e.target.name]: e.target.value })
  }

  render() {
    return (
      <form>
        <input 
          type="text"
          placeholder="Image URL"
          name="img"
          value={this.state.img}
          onChange={(e) => this.updateInputState(e)}
        />
        <input 
          type="text"
          placeholder="Name"
          name="name"
          value={this.state.name}
          onChange={(e) => this.updateInputState(e)}
        />
        <input 
          type="text"
          placeholder="Description"
          name="description"
          value={this.state.description}
          onChange={(e) => this.updateInputState(e)}
        />
        <input 
          type="text"
          placeholder="Price"
          name="price"
          value={this.state.price}
          onChange={(e) => this.updateInputState(e)}
        />
        <button className="Form_button" onClick={(e) => this.props.addPurchase(e, this.state)}>Add Purchase</button>
      </form>
    )
  }
}

export default Form;