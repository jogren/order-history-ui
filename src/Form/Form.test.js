import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;
  let mockPurchase;

  beforeEach(() => {
    mockPurchase = jest.fn();
    wrapper = shallow(<Form addPurchase={mockPurchase}/>)
  })

  it('should keep an up-to-date snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update img state when entering into the imageUrl input', () => {
    wrapper.find('input').at(0).simulate('change', {target: { name: 'img', value: './image' }});
    expect(wrapper.state('img')).toEqual('./image')
  })

  it('should update name state when entering into the name input', () => {
    wrapper.find('input').at(1).simulate('change', { target: { name: 'name', value: 'iPhone' } });
    expect(wrapper.state('name')).toEqual('iPhone')
  })

  it('should update description state when entering into the description input', () => {
    wrapper.find('input').at(2).simulate('change', { target: { name: 'description', value: 'The new iPhone18' } });
    expect(wrapper.state('description')).toEqual('The new iPhone18')
  })

  it('should update price state when entering into the price input', () => {
    wrapper.find('input').at(3).simulate('change', { target: { name: 'price', value: '200' } });
    expect(wrapper.state('price')).toEqual('200')
  })

  it('should fire addPurchase method when the button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockPurchase).toHaveBeenCalled();
  })
})