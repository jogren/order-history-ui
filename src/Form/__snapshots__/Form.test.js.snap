// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should keep an up-to-date snapshot 1`] = `
<form>
  <input
    name="img"
    onChange={[Function]}
    placeholder="Image URL"
    type="text"
    value=""
  />
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Price"
    type="text"
    value=""
  />
  <button
    className="Form_button"
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
